server/conn.go

```diff
@@ -1297,6 +1298,15 @@ func (cc *clientConn) writeChunks(ctx context.Context, rs ResultSet, binary bool
 	req := rs.NewRecordBatch()
 	gotColumnInfo := false
 	for {
+		failpoint.Inject("SleepInwriteChunks", func(val failpoint.Value) {
+			time.Sleep(time.Duration(val.(int)) * time.Millisecond)
+		})
+		//MaxExecDuration() return 0 if there is no limit
+		if rs.MaxExecDuration().Nanoseconds() > 0 &&
+			time.Now().After(rs.StartExecTime().Add(rs.MaxExecDuration())) {
+			return errors.New("Query execution was interrupted, max_execution_time exceeded")
```

Comment:
Is there an standard error code for this type of error in MySQL?

Reply:
If the execution is blocked at line 1311 for a long time, it may exceed the maxExecDuration.

Reply:
> Is there an standard error code for this type of error in MySQL?
The original mysql error code is : "1907: Query execution was interrupted, max_execution_time exceeded.", shall the error code be  defined it in MySQLErrName(which is inside parser)?

Reply:
> If the execution is blocked at line 1311 for a long time, it may exceed the maxExecDuration.
Yes, that's a problem and the current implementation is really a compromise.

I had tried closing the result set on timeout before, but that may cause a panic(due to concurrent call to Next() and Close(), no mutex protection).
If a thread is block on disk/network I/O, closing result set will not work either.
Please let me know if there is any better solution, thanks.

Reply:
It is better to use this predefined error in parser repo,
https://github.com/pingcap/parser/blob/f74198ed55281727f26567edd2fae775bc6c5366/mysql/errname.go#L335

Reply:
> It is better to use this predefined error in parser repo,
> 
> 
OK, I will add and use the error code. 
By the way, should all published Mysql error codes  be imported in batch?

Reply:
Basically, it will be defined when it is used.